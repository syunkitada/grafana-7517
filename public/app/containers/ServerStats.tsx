import React from "react";
import PageHeader from "app/core/components/PageHeader/PageHeader";
import { NavModel, NavModelSrv } from "app/core/nav_model_srv";

export interface IState {
  navModel: NavModel;
}

export default class ServerStats extends React.Component<any, IState> {
  constructor(props) {
    super(props);

    const navModelSrv = new NavModelSrv();

    this.state = {
      navModel: navModelSrv.getNav("cfg", "admin", "server-stats", 1)
    };
  }

  render() {
    return (
      <PageHeader model={this.state.navModel}>
        <h2>ServerStats</h2>
      </PageHeader>
    );
  }
}
