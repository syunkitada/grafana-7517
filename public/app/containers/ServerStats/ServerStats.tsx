import React from 'react';
import { inject, observer } from 'mobx-react';
import PageHeader from 'app/core/components/PageHeader/PageHeader';
import { NavModel, NavModelSrv } from 'app/core/nav_model_srv';

export interface IProps {
  store: any;
}

@inject('store')
@observer
export class ServerStats extends React.Component<IProps, any> {
  constructor(props) {
    super(props);

    // this.navModel = new NavModelSrv().getNav('cfg', 'admin', 'server-stats', 1);
    this.props.store.nav.load('cfg', 'admin', 'server-stats');
    this.props.store.serverStats.load();
  }

  render() {
    return (
      <div>
        <PageHeader model={this.props.store.nav} />
        <div className="page-container page-body">
          <table className="filter-table form-inline">
            <thead>
              <tr>
                <th>Name</th>
                <th>Value</th>
              </tr>
            </thead>
            <tbody>{this.props.store.serverStats.stats.map(StatItem)}</tbody>
          </table>
        </div>
      </div>
    );
  }
}

function StatItem(stat) {
  return (
    <tr key={stat.name}>
      <td>{stat.name}</td>
      <td>{stat.value}</td>
    </tr>
  );
}
