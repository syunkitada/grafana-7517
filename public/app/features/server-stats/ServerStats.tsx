import React from 'react';
import { hot } from 'react-hot-loader';
import { connect } from 'react-redux';
import { initNav } from 'app/core/actions';
import { ContainerProps } from 'app/types';
import PageHeader from 'app/core/components/PageHeader/PageHeader';

interface Props extends ContainerProps {}

export class ServerStats extends React.Component<Props, any> {
  constructor(props) {
    super(props);

    this.props.initNav('cfg', 'admin', 'server-stats');
    // const { nav, serverStats } = this.props;
    //
    // nav.load('cfg', 'admin', 'server-stats');
    // serverStats.load();
    //
    // store.dispatch(setNav('new', { asd: 'tasd' }));
  }

  render() {
    const { navModel } = this.props;
    console.log('render', navModel);
    return (
      <div>
        <PageHeader model={navModel} />
        <h2>aasd</h2>
      </div>
    );
    // const { nav, serverStats } = this.props;
    // return (
    //   <div>
    //     <PageHeader model={nav as any} />
    //     <div className="page-container page-body">
    //       <table className="filter-table form-inline">
    //         <thead>
    //           <tr>
    //             <th>Name</th>
    //             <th>Value</th>
    //           </tr>
    //         </thead>
    //         <tbody>{serverStats.stats.map(StatItem)}</tbody>
    //       </table>
    //     </div>
    //   </div>
    // );
  }
}

function StatItem(stat) {
  return (
    <tr key={stat.name}>
      <td>{stat.name}</td>
      <td>{stat.value}</td>
    </tr>
  );
}

const mapStateToProps = state => ({
  navModel: state.navModel,
});

const mapDispatchToProps = {
  initNav,
};

export default hot(module)(connect(mapStateToProps, mapDispatchToProps)(ServerStats));
