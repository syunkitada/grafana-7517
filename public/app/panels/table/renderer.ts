
export class TableRenderer {
  constructor(private panel, private table) {
  }

  formatColumnValue(columnIndex, value) {
    return "value";
  }

  renderCell(columnIndex, value) {
    var colValue = this.formatColumnValue(columnIndex, value);
    return '<td>' + colValue + '</td>';
  }

  render(page) {
    let endPos = Math.min(this.panel.pageSize, this.table.rows.length);
    let startPos = 0;
    var html = "";

    for (var y = startPos; y < endPos; y++) {
      let row = this.table.rows[y];
      html += '<tr>';
      for (var i = 0; i < this.table.columns.length; i++) {
        html += this.renderCell(i, row[i]);
      }
      html += '</tr>';
    }

    return html;
  }
}
