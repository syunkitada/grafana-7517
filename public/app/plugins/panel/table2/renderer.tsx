// Libraries
import _ from 'lodash';
import moment from 'moment';
import React, { CSSProperties, ReactNode } from 'react';

import { sanitize } from 'app/core/utils/text';

// Types
import kbn from 'app/core/utils/kbn';
import { getValueFormat, getColorFromHexRgbOrName, GrafanaThemeType, InterpolateFunction } from '@grafana/ui';
import { Style } from './types';
import { SortedTableData } from './sortable';
import { Index } from 'react-virtualized';

type CellFormatter = (v: any, style: Style) => string;

interface ColumnInfo {
  header: string;
  accessor: string; // the field name
  style?: Style;
  hidden?: boolean;
  formatter: CellFormatter;
  filterable?: boolean;
}

export class TableRenderer {
  isUTC: false; // TODO? get UTC from props?

  columns: ColumnInfo[];
  colorState: any;
  theme?: GrafanaThemeType;

  constructor(
    styles: Style[],
    data: SortedTableData,
    private rowGetter: (info: Index) => any[], // matches the table rowGetter
    private replaceVariables: InterpolateFunction
  ) {
    this.colorState = {};

    if (!data) {
      this.columns = [];
      return;
    }

    this.columns = data.getInfo().map((col, index) => {
      let title = col.text;
      let style: Style = null;

      for (let i = 0; i < styles.length; i++) {
        const s = styles[i];
        const regex = kbn.stringToJsRegex(s.pattern);
        if (title.match(regex)) {
          style = s;
          if (s.alias) {
            title = title.replace(regex, s.alias);
          }
          break;
        }
      }

      return {
        header: title,
        accessor: col.text, // unique?
        style: style,
        formatter: this.createColumnFormatter(style, col),
      };
    });
  }

  setTheme(theme: GrafanaThemeType) {
    this.theme = theme;
  }

  getColorForValue(value, style: Style) {
    if (!style.thresholds) {
      return null;
    }
    for (let i = style.thresholds.length; i > 0; i--) {
      if (value >= style.thresholds[i - 1]) {
        return getColorFromHexRgbOrName(style.colors[i], this.theme);
      }
    }
    return getColorFromHexRgbOrName(_.first(style.colors), this.theme);
  }

  defaultCellFormatter(v: any, style: Style): string {
    if (v === null || v === void 0 || v === undefined) {
      return '';
    }

    if (_.isArray(v)) {
      v = v.join(', ');
    }

    if (style && style.sanitize) {
      return sanitize(v);
    } else {
      return _.escape(v);
    }
  }

  createColumnFormatter(style: Style, header: any): CellFormatter {
    if (!style) {
      return this.defaultCellFormatter;
    }

    if (style.type === 'hidden') {
      return v => {
        return undefined;
      };
    }

    if (style.type === 'date') {
      return v => {
        if (v === undefined || v === null) {
          return '-';
        }

        if (_.isArray(v)) {
          v = v[0];
        }
        let date = moment(v);
        if (this.isUTC) {
          date = date.utc();
        }
        return date.format(style.dateFormat);
      };
    }

    if (style.type === 'string') {
      return v => {
        if (_.isArray(v)) {
          v = v.join(', ');
        }

        const mappingType = style.mappingType || 0;

        if (mappingType === 1 && style.valueMaps) {
          for (let i = 0; i < style.valueMaps.length; i++) {
            const map = style.valueMaps[i];

            if (v === null) {
              if (map.value === 'null') {
                return map.text;
              }
              continue;
            }

            // Allow both numeric and string values to be mapped
            if ((!_.isString(v) && Number(map.value) === Number(v)) || map.value === v) {
              this.setColorState(v, style);
              return this.defaultCellFormatter(map.text, style);
            }
          }
        }

        if (mappingType === 2 && style.rangeMaps) {
          for (let i = 0; i < style.rangeMaps.length; i++) {
            const map = style.rangeMaps[i];

            if (v === null) {
              if (map.from === 'null' && map.to === 'null') {
                return map.text;
              }
              continue;
            }

            if (Number(map.from) <= Number(v) && Number(map.to) >= Number(v)) {
              this.setColorState(v, style);
              return this.defaultCellFormatter(map.text, style);
            }
          }
        }

        if (v === null || v === void 0) {
          return '-';
        }

        this.setColorState(v, style);
        return this.defaultCellFormatter(v, style);
      };
    }

    if (style.type === 'number') {
      const valueFormatter = getValueFormat(style.unit || header.unit);

      return v => {
        if (v === null || v === void 0) {
          return '-';
        }

        if (_.isString(v) || _.isArray(v)) {
          return this.defaultCellFormatter(v, style);
        }

        this.setColorState(v, style);
        return valueFormatter(v, style.decimals, null);
      };
    }

    return value => {
      return this.defaultCellFormatter(value, style);
    };
  }

  setColorState(value: any, style: Style) {
    if (!style.colorMode) {
      return;
    }

    if (value === null || value === void 0 || _.isArray(value)) {
      return;
    }

    if (_.isNaN(value)) {
      return;
    }
    const numericValue = Number(value);
    this.colorState[style.colorMode] = this.getColorForValue(numericValue, style);
  }

  renderRowVariables(rowIndex: number) {
    const scopedVars = {};
    const row = this.rowGetter({ index: rowIndex });
    for (let i = 0; i < row.length; i++) {
      scopedVars[`__cell_${i}`] = { value: row[i] };
    }
    return scopedVars;
  }

  renderCell(columnIndex: number, rowIndex: number, value: any): ReactNode {
    const column = this.columns[columnIndex];
    if (column.formatter) {
      value = column.formatter(value, column.style);
    }

    const style: CSSProperties = {};
    const cellClasses = [];
    let cellClass = '';

    if (this.colorState.cell) {
      style.backgroundColor = this.colorState.cell;
      style.color = 'white';
      this.colorState.cell = null;
    } else if (this.colorState.value) {
      style.color = this.colorState.value;
      this.colorState.value = null;
    }

    if (value === undefined) {
      style.display = 'none';
      column.hidden = true;
    } else {
      column.hidden = false;
    }

    if (column.style && column.style.preserveFormat) {
      cellClasses.push('table-panel-cell-pre');
    }

    let columnHtml: JSX.Element;
    if (column.style && column.style.link) {
      // Render cell as link
      const scopedVars = this.renderRowVariables(rowIndex);
      scopedVars['__cell'] = { value: value };

      const cellLink = this.replaceVariables(column.style.linkUrl, scopedVars, encodeURIComponent);
      const cellLinkTooltip = this.replaceVariables(column.style.linkTooltip, scopedVars);
      const cellTarget = column.style.linkTargetBlank ? '_blank' : '';

      cellClasses.push('table-panel-cell-link');
      columnHtml = (
        <a
          href={cellLink}
          target={cellTarget}
          data-link-tooltip
          data-original-title={cellLinkTooltip}
          data-placement="right"
        >
          {value}
        </a>
      );
    } else {
      columnHtml = <span>{value}</span>;
    }

    let filterLink: JSX.Element;
    if (column.filterable) {
      cellClasses.push('table-panel-cell-filterable');
      filterLink = (
        <span>
          <a
            className="table-panel-filter-link"
            data-link-tooltip
            data-original-title="Filter out value"
            data-placement="bottom"
            data-row={rowIndex}
            data-column={columnIndex}
            data-operator="!="
          >
            <i className="fa fa-search-minus" />
          </a>
          <a
            className="table-panel-filter-link"
            data-link-tooltip
            data-original-title="Filter for value"
            data-placement="bottom"
            data-row={rowIndex}
            data-column={columnIndex}
            data-operator="="
          >
            <i className="fa fa-search-plus" />
          </a>
        </span>
      );
    }

    if (cellClasses.length) {
      cellClass = cellClasses.join(' ');
    }

    style.width = '100%';
    style.height = '100%';
    columnHtml = (
      <div className={cellClass} style={style}>
        {columnHtml}
        {filterLink}
      </div>
    );
    return columnHtml;
  }
}
