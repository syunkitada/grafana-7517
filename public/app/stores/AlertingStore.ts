import { types, getEnv, flow } from 'mobx-state-tree';
import moment from 'moment';
import alertDef from 'app/features/alerting/alert_def';

export const AlertRule = types.model('AlertRule', {
  id: types.identifier(types.number),
  dashboardId: types.number,
  panelId: types.number,
  name: types.string,
  state: types.string,
  stateText: types.string,
  stateIcon: types.string,
  stateClass: types.string,
  stateAge: types.string,
  info: types.optional(types.string, ''),
  dashboardUri: types.string,
});

export const AlertingStore = types
  .model('AlertingStore', {
    rules: types.array(AlertRule),
  })
  .actions(self => ({
    loadRules: flow(function* load() {
      let backendSrv = getEnv(self).backendSrv;

      let rules = yield backendSrv.get('/api/alerts');

      self.rules.clear();

      for (let rule of rules) {
        let stateModel = alertDef.getStateDisplayModel(rule.state);
        rule.stateText = stateModel.text;
        rule.stateIcon = stateModel.iconClass;
        rule.stateClass = stateModel.stateClass;
        rule.stateAge = moment(rule.newStateDate)
          .fromNow()
          .replace(' ago', '');

        if (rule.executionError) {
          rule.info = 'Execution Error: ' + rule.executionError;
        }

        if (rule.evalData && rule.evalData.noData) {
          rule.info = 'Query returned no data';
        }

        self.rules.push(AlertRule.create(rule));
      }
    }),
  }));
