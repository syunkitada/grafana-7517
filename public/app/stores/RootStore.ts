import { types } from 'mobx-state-tree';
import { SearchStore } from './SearchStore';
import { ServerStatsStore } from './ServerStatsStore';
import { NavStore } from './NavStore/NavStore';

export const RootStore = types.model({
  search: types.optional(SearchStore, {
    sections: [],
  }),
  serverStats: types.optional(ServerStatsStore, {
    stats: [],
  }),
  nav: types.optional(NavStore, {}),
});

type IRootStoreType = typeof RootStore.Type;
export interface IRootStore extends IRootStoreType {}
