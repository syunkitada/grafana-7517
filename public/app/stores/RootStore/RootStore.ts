import { types } from 'mobx-state-tree';
import { NavStore } from './../NavStore/NavStore';
import { ViewStore } from './../ViewStore/ViewStore';
import { PermissionsStore } from './../PermissionsStore/PermissionsStore';

export const RootStore = types.model({
  nav: types.optional(NavStore, {}),
  permissions: types.optional(PermissionsStore, {
    fetching: false,
    items: [],
  }),
  view: types.optional(ViewStore, {
    path: '',
    query: {},
    routeParams: {},
  }),
});

type RootStoreType = typeof RootStore.Type;
export interface RootStoreInterface extends RootStoreType {}
