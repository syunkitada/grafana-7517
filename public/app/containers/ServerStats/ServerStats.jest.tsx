import React from 'react';
import renderer from 'react-test-renderer';
import { ServerStats } from './ServerStats';
import { RootStore } from 'app/stores/RootStore';

describe('ServerStats', () => {
  it('Should render table with stats', done => {
    let backendSrvMock = {
      get: jest.fn().mockReturnValue(
        Promise.resolve({
          dashboards: 10,
        })
      ),
    };

    const store = RootStore.create({}, { backendSrv: backendSrvMock });
    const page = renderer.create(<ServerStats store={store} />);

    setTimeout(() => {
      expect(page.toJSON()).toMatchSnapshot();
      done();
    });
  });
});
