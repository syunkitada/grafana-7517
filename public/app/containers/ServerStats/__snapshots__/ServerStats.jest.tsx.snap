// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerStats Should render table with stats 1`] = `
<div>
  // 
  <div
    className="page-container page-body"
  >
    <table
      className="filter-table form-inline"
    >
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Value
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Total dashboards
          </td>
          <td>
            10
          </td>
        </tr>
        <tr>
          <td>
            Total users
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            Active users (seen last 30 days)
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            Total orgs
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            Total playlists
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            Total snapshots
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            Total dashboard tags
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            Total starred dashboards
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            Total alerts
          </td>
          <td>
            0
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
